proof/crefine/RISCV64/Ipc_C.thy

```diff
@@ -4014,6 +4150,209 @@ lemma copyMRsFaultReply_ccorres_syscall:
   done
 qed
 
+lemma copyMRsFaultReply_ccorres_timeout:
+  fixes word_size :: "'a::len"
+  shows "ccorres dc xfdc
+           (valid_pspace' and tcb_at' s
+                          and tcb_at' r
+                          and obj_at' (\<lambda>t. tcbFault t = Some f) r
+                          and K (r \<noteq> s)
+                          and K (len \<le> unat n_timeoutMessage))
+           (UNIV \<inter> \<lbrace>\<acute>sender = tcb_ptr_to_ctcb_ptr s\<rbrace>
+                 \<inter> \<lbrace>\<acute>receiver = tcb_ptr_to_ctcb_ptr r\<rbrace>
+                 \<inter> \<lbrace>\<acute>id___anonymous_enum = MessageID_TimeoutReply \<rbrace>
+                 \<inter> \<lbrace>\<acute>length___unsigned_long = of_nat len \<rbrace>) hs
+           (do t \<leftarrow> getSanitiseRegisterInfo r;
+               a \<leftarrow> zipWithM_x (\<lambda>rs rd. do v \<leftarrow> asUser s (getRegister rs);
+                                           asUser r (setRegister rd (RISCV64_H.sanitiseRegister t rd v))
+                                        od)
+               RISCV64_H.msgRegisters (take len RISCV64_H.timeoutMessage);
+               sendBuf \<leftarrow> lookupIPCBuffer False s;
+               case sendBuf of None \<Rightarrow> return ()
+                            | Some bufferPtr \<Rightarrow>
+                                   zipWithM_x (\<lambda>i rd. do v \<leftarrow> loadWordUser (bufferPtr + i * 8);
+                                                         asUser r (setRegister rd (RISCV64_H.sanitiseRegister t rd v))
+                                                      od)
+                                      [scast n_msgRegisters + 1.e.scast n_timeoutMessage]
+                                      (drop (unat (scast n_msgRegisters :: machine_word))
+                                            (take len RISCV64_H.timeoutMessage))
+                                 od)
+           (Call copyMRsFaultReply_'proc)"
+  proof -
+  let ?obj_at_ft = "obj_at' (\<lambda>tcb. tcbFault tcb = Some f) s"
+  note symb_exec_r_fault = ccorres_symb_exec_r_known_rv_UNIV
+          [where xf'=ret__unsigned_' and R="?obj_at_ft" and R'=UNIV]
+  note empty_fail_cond[simp]
+  show ?thesis
```

Comment:
Can't tell from the GitHub view, but either the `proof -` lines is indented too much, or the rest is not indented by 2 when it should be.